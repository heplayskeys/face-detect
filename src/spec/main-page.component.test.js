import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import MainPage from '../components/main-page/main-page.component';

let wrapper;

describe('<MainPage />', () => {
	beforeEach(() => {
		const mockProps = {
			input: '',
			route: 'signin',
			activeUser: null,
			detectedFaces: [],
			isPending: false,
			setInput: jest.fn(),
			setRoute: jest.fn(),
			setUser: jest.fn(),
			updateUser: jest.fn(),
			getFaces: jest.fn()
		};
		wrapper = shallow(<MainPage {...mockProps} />);
	});

	it('renders main page component successfully', () => {
		expect(wrapper).toMatchSnapshot();
	});

	it('renders sign in and register pages from setPage function', () => {
		expect(shallowToJson(wrapper)).toMatchSnapshot();
		wrapper.setProps({ route: 'register' });
		expect(shallowToJson(wrapper)).toMatchSnapshot();
	});

	it('renders main page component with signed in user and sign out', () => {
		wrapper.setProps({ activeUser: {}, route: 'home' });
		expect(shallowToJson(wrapper)).toMatchSnapshot();
		wrapper.setProps({ activeUser: null, route: 'signin' });
		expect(shallowToJson(wrapper)).toMatchSnapshot();
	});
});
