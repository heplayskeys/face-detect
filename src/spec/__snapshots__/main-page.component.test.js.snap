// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MainPage /> renders main page component successfully 1`] = `ShallowWrapper {}`;

exports[`<MainPage /> renders main page component with signed in user and sign out 1`] = `
<div
  className="App"
>
  <f
    className="particles"
    height="100%"
    id="tsparticles"
    params={
      Object {
        "interactivity": Object {
          "detect_on": "canvas",
          "events": Object {
            "onclick": Object {
              "enable": true,
              "mode": "push",
            },
            "onhover": Object {
              "enable": true,
              "mode": "repulse",
            },
            "resize": true,
          },
          "modes": Object {
            "bubble": Object {
              "distance": 400,
              "duration": 2,
              "opacity": 8,
              "size": 40,
              "speed": 3,
            },
            "grab": Object {
              "distance": 400,
              "line_linked": Object {
                "opacity": 1,
              },
            },
            "push": Object {
              "particles_nb": 4,
            },
            "remove": Object {
              "particles_nb": 2,
            },
            "repulse": Object {
              "distance": 200,
              "duration": 0.4,
            },
          },
        },
        "particles": Object {
          "color": Object {
            "value": "#ffffff",
          },
          "line_linked": Object {
            "color": "#ffffff",
            "distance": 175,
            "enable": true,
            "opacity": 0.2963200256276779,
            "width": 1,
          },
          "move": Object {
            "attract": Object {
              "enable": false,
              "rotateX": 600,
              "rotateY": 1200,
            },
            "bounce": false,
            "direction": "none",
            "enable": true,
            "out_mode": "bounce",
            "random": true,
            "speed": 3,
            "straight": false,
          },
          "number": Object {
            "density": Object {
              "enable": true,
              "value_area": 800,
            },
            "value": 85,
          },
          "opacity": Object {
            "anim": Object {
              "enable": true,
              "opacity_min": 0.1,
              "speed": 1,
              "sync": false,
            },
            "random": true,
            "value": 0.49716301422833176,
          },
          "shape": Object {
            "image": Object {
              "height": 100,
              "src": "img/github.svg",
              "width": 100,
            },
            "polygon": Object {
              "nb_sides": 5,
            },
            "stroke": Object {
              "color": "#ffffff",
              "width": 0,
            },
            "type": "circle",
          },
          "size": Object {
            "anim": Object {
              "enable": true,
              "size_min": 0.1,
              "speed": 9.73386004331568,
              "sync": false,
            },
            "random": true,
            "value": 3,
          },
        },
        "retina_detect": true,
      }
    }
    style={Object {}}
    width="100%"
  />
  <Connect(Navigation)
    route="home"
  />
  <div>
    <Logo />
    <UserGreeting />
    <Connect(Rank) />
    <Connect(ImageLinkForm)
      URL="https://boiling-brushlands-70070.herokuapp.com"
      handleSubmit={[Function]}
      input=""
      numFaces={0}
      recentGrabs={Array []}
    />
    <FaceDetect
      detectedFaces={Array []}
      error=""
      imageURL=""
    />
  </div>
</div>
`;

exports[`<MainPage /> renders main page component with signed in user and sign out 2`] = `
<div
  className="App"
>
  <f
    className="particles"
    height="100%"
    id="tsparticles"
    params={
      Object {
        "interactivity": Object {
          "detect_on": "canvas",
          "events": Object {
            "onclick": Object {
              "enable": true,
              "mode": "push",
            },
            "onhover": Object {
              "enable": true,
              "mode": "repulse",
            },
            "resize": true,
          },
          "modes": Object {
            "bubble": Object {
              "distance": 400,
              "duration": 2,
              "opacity": 8,
              "size": 40,
              "speed": 3,
            },
            "grab": Object {
              "distance": 400,
              "line_linked": Object {
                "opacity": 1,
              },
            },
            "push": Object {
              "particles_nb": 4,
            },
            "remove": Object {
              "particles_nb": 2,
            },
            "repulse": Object {
              "distance": 200,
              "duration": 0.4,
            },
          },
        },
        "particles": Object {
          "color": Object {
            "value": "#ffffff",
          },
          "line_linked": Object {
            "color": "#ffffff",
            "distance": 175,
            "enable": true,
            "opacity": 0.2963200256276779,
            "width": 1,
          },
          "move": Object {
            "attract": Object {
              "enable": false,
              "rotateX": 600,
              "rotateY": 1200,
            },
            "bounce": false,
            "direction": "none",
            "enable": true,
            "out_mode": "bounce",
            "random": true,
            "speed": 3,
            "straight": false,
          },
          "number": Object {
            "density": Object {
              "enable": true,
              "value_area": 800,
            },
            "value": 85,
          },
          "opacity": Object {
            "anim": Object {
              "enable": true,
              "opacity_min": 0.1,
              "speed": 1,
              "sync": false,
            },
            "random": true,
            "value": 0.49716301422833176,
          },
          "shape": Object {
            "image": Object {
              "height": 100,
              "src": "img/github.svg",
              "width": 100,
            },
            "polygon": Object {
              "nb_sides": 5,
            },
            "stroke": Object {
              "color": "#ffffff",
              "width": 0,
            },
            "type": "circle",
          },
          "size": Object {
            "anim": Object {
              "enable": true,
              "size_min": 0.1,
              "speed": 9.73386004331568,
              "sync": false,
            },
            "random": true,
            "value": 3,
          },
        },
        "retina_detect": true,
      }
    }
    style={Object {}}
    width="100%"
  />
  <Connect(Navigation)
    route="signin"
  />
  <Connect(SignIn)
    URL="https://boiling-brushlands-70070.herokuapp.com"
  />
</div>
`;

exports[`<MainPage /> renders sign in and register pages from setPage function 1`] = `
<div
  className="App"
>
  <f
    className="particles"
    height="100%"
    id="tsparticles"
    params={
      Object {
        "interactivity": Object {
          "detect_on": "canvas",
          "events": Object {
            "onclick": Object {
              "enable": true,
              "mode": "push",
            },
            "onhover": Object {
              "enable": true,
              "mode": "repulse",
            },
            "resize": true,
          },
          "modes": Object {
            "bubble": Object {
              "distance": 400,
              "duration": 2,
              "opacity": 8,
              "size": 40,
              "speed": 3,
            },
            "grab": Object {
              "distance": 400,
              "line_linked": Object {
                "opacity": 1,
              },
            },
            "push": Object {
              "particles_nb": 4,
            },
            "remove": Object {
              "particles_nb": 2,
            },
            "repulse": Object {
              "distance": 200,
              "duration": 0.4,
            },
          },
        },
        "particles": Object {
          "color": Object {
            "value": "#ffffff",
          },
          "line_linked": Object {
            "color": "#ffffff",
            "distance": 175,
            "enable": true,
            "opacity": 0.2963200256276779,
            "width": 1,
          },
          "move": Object {
            "attract": Object {
              "enable": false,
              "rotateX": 600,
              "rotateY": 1200,
            },
            "bounce": false,
            "direction": "none",
            "enable": true,
            "out_mode": "bounce",
            "random": true,
            "speed": 3,
            "straight": false,
          },
          "number": Object {
            "density": Object {
              "enable": true,
              "value_area": 800,
            },
            "value": 85,
          },
          "opacity": Object {
            "anim": Object {
              "enable": true,
              "opacity_min": 0.1,
              "speed": 1,
              "sync": false,
            },
            "random": true,
            "value": 0.49716301422833176,
          },
          "shape": Object {
            "image": Object {
              "height": 100,
              "src": "img/github.svg",
              "width": 100,
            },
            "polygon": Object {
              "nb_sides": 5,
            },
            "stroke": Object {
              "color": "#ffffff",
              "width": 0,
            },
            "type": "circle",
          },
          "size": Object {
            "anim": Object {
              "enable": true,
              "size_min": 0.1,
              "speed": 9.73386004331568,
              "sync": false,
            },
            "random": true,
            "value": 3,
          },
        },
        "retina_detect": true,
      }
    }
    style={Object {}}
    width="100%"
  />
  <Connect(Navigation)
    route="signin"
  />
  <Connect(SignIn)
    URL="https://boiling-brushlands-70070.herokuapp.com"
  />
</div>
`;

exports[`<MainPage /> renders sign in and register pages from setPage function 2`] = `
<div
  className="App"
>
  <f
    className="particles"
    height="100%"
    id="tsparticles"
    params={
      Object {
        "interactivity": Object {
          "detect_on": "canvas",
          "events": Object {
            "onclick": Object {
              "enable": true,
              "mode": "push",
            },
            "onhover": Object {
              "enable": true,
              "mode": "repulse",
            },
            "resize": true,
          },
          "modes": Object {
            "bubble": Object {
              "distance": 400,
              "duration": 2,
              "opacity": 8,
              "size": 40,
              "speed": 3,
            },
            "grab": Object {
              "distance": 400,
              "line_linked": Object {
                "opacity": 1,
              },
            },
            "push": Object {
              "particles_nb": 4,
            },
            "remove": Object {
              "particles_nb": 2,
            },
            "repulse": Object {
              "distance": 200,
              "duration": 0.4,
            },
          },
        },
        "particles": Object {
          "color": Object {
            "value": "#ffffff",
          },
          "line_linked": Object {
            "color": "#ffffff",
            "distance": 175,
            "enable": true,
            "opacity": 0.2963200256276779,
            "width": 1,
          },
          "move": Object {
            "attract": Object {
              "enable": false,
              "rotateX": 600,
              "rotateY": 1200,
            },
            "bounce": false,
            "direction": "none",
            "enable": true,
            "out_mode": "bounce",
            "random": true,
            "speed": 3,
            "straight": false,
          },
          "number": Object {
            "density": Object {
              "enable": true,
              "value_area": 800,
            },
            "value": 85,
          },
          "opacity": Object {
            "anim": Object {
              "enable": true,
              "opacity_min": 0.1,
              "speed": 1,
              "sync": false,
            },
            "random": true,
            "value": 0.49716301422833176,
          },
          "shape": Object {
            "image": Object {
              "height": 100,
              "src": "img/github.svg",
              "width": 100,
            },
            "polygon": Object {
              "nb_sides": 5,
            },
            "stroke": Object {
              "color": "#ffffff",
              "width": 0,
            },
            "type": "circle",
          },
          "size": Object {
            "anim": Object {
              "enable": true,
              "size_min": 0.1,
              "speed": 9.73386004331568,
              "sync": false,
            },
            "random": true,
            "value": 3,
          },
        },
        "retina_detect": true,
      }
    }
    style={Object {}}
    width="100%"
  />
  <Connect(Navigation)
    route="register"
  />
  <Connect(Register)
    URL="https://boiling-brushlands-70070.herokuapp.com"
  />
</div>
`;
